### PR TITLE
Avoid adding two Filters of different DF from one Attribute

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-4604

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

File: libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts [hunk 1]
```diff
@@ -27,5 +27,9 @@ export async function canFilterBeAdded(
         loadExistingDisplayForms,
     ]);
 
-    return !existingDisplayForms.some((existing) => areObjRefsEqual(existing, addedDisplayForm));
+    const attributeFilterRef = addedDisplayForm.attribute;
+    const existingAttributes = existingDisplayForms.map((df) => df.attribute);
+
+    // lookup has to be throw related Attributes since we support only one DF for one Attribute
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts:33]: This comment is confusing, it talks about a throw but it does not throw anything.

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-dashboard/src/model/commandHandlers/filterContext/attributeFilter/validation/uniqueFiltersValidation.ts]: fixed

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR3116

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2956/ : FAILURE in 3m 00s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/3263/ : SUCCESS in 29s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/2110/ : SUCCESS in 23m 51s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/12236/ : SUCCESS in 12m 49s

**Pull request merged**